### PR TITLE
rgw: Add documentation for the Multi-tenancy feature

### PR DESCRIPTION
**Pull request by zaitcev**
This is the user documentation for the Multi-tenancy, including simple examples.

### CODE REVIEW

**Comment by mattbenjamin** [comment]
@zaitcev @yehudasa what will it take to remove DNM? :)

**Comment by zaitcev** [comment]
@mattbenjamin Did you review the document? Do you think that in-tree delivery method as .rst is appropriate and the content is useful to users of Ceph? If yes, then I can drop that DNM in a minute.

File: doc/radosgw/multitenancy.rst [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+create buckets with easily conflicting names such as "test" or "trove".
+
+From the Jewel release onward, each user and bucket lies under a tenant.
+For compatibility, a "legacy" tenant with an empty name is provided.
+Whenever a bucket is referred without an explicit tenant, an implicit
+tenant is used, taken from the user performing the operation. Since
+the pre-existing users are under the legacy tenant, they continue
+to create and access buckets as before. The layout of objects in RADOS
+is extended in a compatible way, ensuring a smooth upgrade to Jewel
+(see "layout.rst" for details XXX).
```
Comment (mattbenjamin) [inline doc/radosgw/multitenancy.rst]: what does the XXX point to? :)

**Comment by mattbenjamin** [comment]
@zaitcev I have (re) reviewed the doc.  I think we should merge it;  @yehudasa do you have any objection?

**Comment by yehudasa** [comment]
@mattbenjamin @zaitcev maybe get rid of the XXX before merging?

**Comment by zaitcev** [comment]
@yehudasa I agree, I'll get that XXX resolved.

**Pull request merged**